python-checks/src/test/resources/checks/cdk/unrestrictedOutboundCommunicationsCheck.py

```diff
@@ -0,0 +1,49 @@
+import aws_cdk.aws_ec2 as ec2
+
+ec2.SecurityGroup(
+    self,
+    "SensitiveExplicit",
+    "sg-1234",
+    allow_all_outbound=True  # NonCompliant{{Make sure that allowing unrestricted outbound communications is safe here.}}
+)
+
+ec2.SecurityGroup(  # NonCompliant{{Omitting "allow_all_outbound" enables unrestricted outbound communications. Make sure it is safe here.}}
+    self,
+    "SensitiveDefault",
+    vpc=vpc
+)
+
+ec2.SecurityGroup.from_security_group_id(
+    self,
+    "SensitiveExplicit",
+    "sg-1234",
+    allow_all_outbound=True  # NonCompliant{{Make sure that allowing unrestricted outbound communications is safe here.}}
+)
+
+ec2.SecurityGroup.from_security_group_id(  # NonCompliant{{Omitting "allow_all_outbound" enables unrestricted outbound communications. Make sure it is safe here.}}
+    self,
+    "SensitiveDefault",
+    vpc=vpc
+)
+
+outbound_restriction=True
```

Comment:
Using ExpressionFlow for arguments you could also highlight the assignment of the sensitive value as a secondary location.